apisix/init.lua

```diff
@@ -893,7 +906,16 @@ function _M.http_balancer_phase()
         return core.response.exit(500)
     end
 
-    load_balancer.run(api_ctx.matched_route, api_ctx, common_phase)
+    if api_ctx.custom_upstream_ip then
+        local ok, err = balancer.set_current_peer(api_ctx.custom_upstream_ip,
+                                                  api_ctx.custom_upstream_port)
+        if not ok then
+            core.log.error("failed to overwrite upstream for ai_proxy: ", err)
+            return core.response.exit(500)
+        end
+    else
```

Comment:
Why not use the `upstream.set` API to modify the upstream endpoints instead of doing this? Is this mandatory?

Reply:
done